File: colorPickerView.js
const { ActionRowBuilder, ButtonBuilder, ButtonStyle, Colors } = require('discord.js');

class ColorPickerView {
    constructor(parentView) {
        this.parent = parentView;
        this.components = this.createColorButtons();
    }

    createColorButtons() {
        const colors = [
            ["Red", Colors.Red],
            ["Blue", Colors.Blue],
            ["Green", Colors.Green],
            ["Purple", Colors.Purple],
            ["Gold", Colors.Yellow], // Discord.js uses Yellow for gold
            ["Orange", Colors.Orange]
        ];

        const rows = [];
        let currentRow = [];

        colors.forEach(([name, color], i) => {
            const button = new ButtonBuilder()
                .setCustomId(`color_${name.toLowerCase()}`)
                .setLabel(name)
                .setStyle(ButtonStyle.Secondary);

            currentRow.push(button);

            if (currentRow.length === 4 || i === colors.length - 1) {
                rows.push(new ActionRowBuilder().addComponents(currentRow));
                currentRow = [];
            }
        });

        return rows;
    }

    async handleInteraction(interaction) {
        const colorName = interaction.customId.split('_')[1];
        const colorMap = {
            'red': Colors.Red,
            'blue': Colors.Blue,
            'green': Colors.Green,
            'purple': Colors.Purple,
            'gold': Colors.Yellow,
            'orange': Colors.Orange
        };

        this.parent.embedData.color = colorMap[colorName];
        await this.parent.updatePreview(interaction);
        await interaction.message.delete();
    }
}

module.exports = ColorPickerView;

File: colorpickerview.py
import discord

class ColorPickerView(discord.ui.View):
    def __init__(self, parent_view):
        super().__init__(timeout=60)
        self.parent = parent_view
        self.add_color_buttons()
    
    def add_color_buttons(self):
        colors = [
            ("Red", discord.Color.red()),
            ("Blue", discord.Color.blue()),
            ("Green", discord.Color.green()),
            ("Purple", discord.Color.purple()),
            ("Gold", discord.Color.gold()),
            ("Orange", discord.Color.orange())
        ]
        
        for i, (name, color) in enumerate(colors):
            button = discord.ui.Button(
                label=name,
                style=discord.ButtonStyle.secondary,
                row=i // 4
            )
            button.callback = self.create_color_callback(color)
            self.add_item(button)
    
    def create_color_callback(self, color):
        async def callback(interaction: discord.Interaction):
            self.parent.embed_data['color'] = color
            await self.parent.update_preview(interaction)
            await interaction.message.delete()
        return callback